test/monitoringlocationlinter/prometheushandler/handler.go

```diff
@@ -0,0 +1,59 @@
+package prometheushandler
+
+import "go/ast"
+
+const (
+	prometheusImportPath = `"github.com/prometheus/client_golang/prometheus"`
+)
+
+// Handler provides different handling, depend on the way prometheus was imported, whether
+// is imported with "." name, custom name or without any name.
+type Handler interface {
+	// IsRegisterMetricsCall checks if a call is a register metrics call
+	IsRegisterMetricsCall(*ast.CallExpr) bool
+}
+
+// GetPrometheusHandler returns a prometheus handler according to the way prometheus was imported in the file.
+func GetPrometheusHandler(file *ast.File) Handler {
+	for _, imp := range file.Imports {
+		if imp.Path.Value != prometheusImportPath {
+			continue
+		}
+
+		switch name := imp.Name.String(); {
+		case name == ".":
+			return dotHandler{}
+		case name == "<nil>": // import with no local name
+			return nameHandler("prometheus")
+		default:
+			return nameHandler(name)
+		}
+	}
+
+	return nil // no prometheus import; this file does not use prometheus
+}
+
+type dotHandler struct{}
+
+func (h dotHandler) IsRegisterMetricsCall(call *ast.CallExpr) bool {
+	if ident, ok := call.Fun.(*ast.Ident); ok {
+		if ident.Name == "Register" || ident.Name == "MustRegister" {
```

Comment:
Maybe add a helper func that get string and return bool, so you won't have to duplicate thid check below.

Reply:
No duplication after removing the handler.